From 940dc7820ac897ed21f6dd0fae73f5f2117f8cbc Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Thu, 01 Feb 2024 10:52:21 +0200
Subject: [PATCH] [SW-172371] gaudi3: update HBW drain timeout config

As per arch team estimated max HBW latency is ~2500 nsec. Taking margin of
10x it is ~25,000 nsec.
The PLL clock is 1GHz (25,000 = 0x61A8 clock cycles)
so modify currently configured value of 0x1000 clock cycles with 0x8000.

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I5a86d04353ea3f3a5b1c9d7abc1da44bf6c57cf3
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 1fdb5b4..0af9a4c 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -304,7 +304,7 @@
 	regmap_writel(PSOC_ARC2_PRELOADED_ID, mmD0_PARC_ARC2_CFG_BASE + mmARC_ARC_NUM);
 }
 
-#define HBW_DRAIN_TIMEOUT	0x1000
+#define HBW_DRAIN_TIMEOUT	0x8000
 #define LBW_DRAIN_TIMEOUT	0xF000
 
 void ppboot_pcie_drain_init(void)
